rsc.js

```diff
@@ -1,47 +1,27 @@
 // @ts-check
 
-// cjs-module-lexer will let all of react's (named) exports through unchanged.
-module.exports = { ...require("react") };
+// missing functions
+module.exports.createContext = polyfillMissingFn("createContext");
+module.exports.createFactory = polyfillMissingFn("createFactory");
+module.exports.act = polyfillMissingFn("act");
+module.exports.unstable_act = polyfillMissingFn("unstable_act");
+module.exports.unstable_useCacheRefresh = polyfillMissingFn("unstable_useCacheRefresh");
+module.exports.useContext = polyfillMissingFn("useContext");
+module.exports.useDeferredValue = polyfillMissingFn("useDeferredValue");
+module.exports.useEffect = polyfillMissingFn("useEffect");
+module.exports.useImperativeHandle = polyfillMissingFn("useImperativeHandle");
+module.exports.useInsertionEffect = polyfillMissingFn("useInsertionEffect");
+module.exports.useLayoutEffect = polyfillMissingFn("useLayoutEffect");
+module.exports.useReducer = polyfillMissingFn("useReducer");
+module.exports.useRef = polyfillMissingFn("useRef");
+module.exports.useState = polyfillMissingFn("useState");
+module.exports.useSyncExternalStore = polyfillMissingFn("useSyncExternalStore");
+module.exports.useTransition = polyfillMissingFn("useTransition");
+module.exports.useOptimistic = polyfillMissingFn("useOptimistic");
 
-const missingFunctions = [
-  "createContext",
-  "createFactory",
-  "act",
-  "unstable_act",
-  "unstable_useCacheRefresh",
-  "useContext",
-  "useDeferredValue",
-  "useEffect",
-  "useImperativeHandle",
-  "useInsertionEffect",
-  "useLayoutEffect",
-  "useReducer",
-  "useRef",
-  "useState",
-  "useSyncExternalStore",
-  "useTransition",
-  "useOptimistic",
-];
-
-const missingClasses = ["Component", "PureComponent"];
-
-missingFunctions.forEach((exportName) => {
-  module.exports[exportName] ||= function throwOnNonExistentImport() {
-    throw new Error(
-      `React functionality '${exportName}' is not available in this environment.`
-    );
-  };
-});
-
-missingClasses.forEach((exportName) => {
-  module.exports[exportName] ||= class NonExistentClass {
-    constructor() {
-      throw new Error(
-        `React class '${exportName}' is not available in this environment.`
-      );
-    }
-  };
-});
+// missing classes
+module.exports.Component = polyfillMissingClass("Component");
+module.exports.PureComponent = polyfillMissingClass("PureComponent");
 
 module.exports.createContext ||= function unsupportedCreateContext() {
```

Comment:
for consistency? if we're not doing it above, we probably don't wanna do it here
```suggestion
module.exports.createContext = function unsupportedCreateContext() {
```

Reply:
I had the same thought - I think I already fixed that up while you were writing this comment :D